components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/service/WebSubHubAdapterServiceImpl.java

```diff
@@ -0,0 +1,329 @@
+                log.error("WebHubSub event publisher received " + responseCode + " code. Response data: " +
+                        errorResponseBody);
+            } catch (IOException e) {
+                log.error("Error while reading WebSubHub event publisher response. ", e);
+            }
+        }
+    }
+
+    private static void handleSuccessfulTopicMgt(CloseableHttpResponse response, HttpPost httpPost, String topic,
+                                                 String operation, long requestStartTime, int responseCode,
+                                                 String responsePhrase) throws IOException, WebSubAdapterException {
+
+        HttpEntity entity = response.getEntity();
+        WebSubHubCorrelationLogUtils.triggerCorrelationLogForResponse(httpPost, requestStartTime,
+                WebSubHubCorrelationLogUtils.RequestStatus.COMPLETED.getStatus(),
+                String.valueOf(responseCode), responsePhrase);
+
+        if (entity != null) {
+            String responseString = EntityUtils.toString(entity, StandardCharsets.UTF_8);
+            if (RESPONSE_FOR_SUCCESSFUL_OPERATION.equals(responseString)) {
+                log.debug("Success WebSub Hub operation: " + operation + ", topic: " + topic);
+            } else {
+                throw handleServerException(ERROR_INVALID_RESPONSE_FROM_WEBSUB_HUB, null,
+                        topic, operation, responseString);
+            }
+        } else {
+            String message = String.format(ERROR_EMPTY_RESPONSE_FROM_WEBSUB_HUB.getDescription(), topic, operation);
+            throw new WebSubAdapterServerException(message, ERROR_EMPTY_RESPONSE_FROM_WEBSUB_HUB.getCode());
+        }
+    }
+
+    private static void handleConflictOrNotFound(CloseableHttpResponse response, HttpPost httpPost, String topic,
+                                                 String operation, long requestStartTime,
+                                                 int responseCode, String responsePhrase) throws IOException {
+
+        HttpEntity entity = response.getEntity();
+        String responseString = "";
+        WebSubHubCorrelationLogUtils.triggerCorrelationLogForResponse(httpPost, requestStartTime,
+                WebSubHubCorrelationLogUtils.RequestStatus.FAILED.getStatus(),
+                String.valueOf(responseCode), responsePhrase);
+        if (entity != null) {
+            responseString = EntityUtils.toString(entity, StandardCharsets.UTF_8);
+        }
+        log.warn(String.format(ERROR_INVALID_RESPONSE_FROM_WEBSUB_HUB.getDescription(),
+                topic, operation, responseString));
+    }
+
+    private static void handleFailedTopicMgt(CloseableHttpResponse response, HttpPost httpPost, String topic,
+                                             String operation, long requestStartTime, int responseCode,
+                                             String responsePhrase) throws IOException, WebSubAdapterException {
```

Comment:
It's not a good practice pass so many variables as inputs.
Recommended practice is to have like four. If things go beyond that means there's a problem in code

Reply:
Addressed with https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/7999427bc5211016a93107fb27f9efd67513d525